packages/client/test/events.spec.ts

```diff
@@ -569,15 +582,93 @@ describe('Events', () => {
         it('Stale', (done) => {
           const provider = new MockProvider({ initialStatus: ProviderStatus.STALE });
           const client = OpenFeature.getClient(clientId);
-  
+
           OpenFeature.setProvider(clientId, provider);
           expect(provider.initialize).not.toHaveBeenCalled();
-  
+
           client.addHandler(ProviderEvents.Stale, () => {
             done();
           });
         });
       });
     });
   });
+
+  describe('Requirement 5.3.4.2, 5.3.4.3', () => {
+    describe('API', () => {
+      describe('context set for same client', () => {
+        it("If the provider's `on context changed` function terminates normally, associated `PROVIDER_CONTEXT_CHANGED` handlers MUST run.", (done) => {
+          const provider = new MockProvider({ initialStatus: ProviderStatus.READY });
+  
+          OpenFeature.setProvider(clientId, provider);
+          OpenFeature.setContext(clientId, {});
+  
+          OpenFeature.addHandler(ProviderEvents.ContextChanged, () => {
+            done();
+          });
+        });
+  
+        it("If the provider's `on context changed` function terminates abnormally, associated `PROVIDER_ERROR` handlers MUST run.", (done) => {
+          const provider = new MockProvider({ initialStatus: ProviderStatus.READY, failOnContextChange: true });
+  
+          OpenFeature.setProvider(clientId, provider);
+          OpenFeature.setContext(clientId, {});
+  
+          OpenFeature.addHandler(ProviderEvents.Error, () => {
+            done();
+          });
+        });
+      });
+      
+      describe('context set for different client', () => {
+        it("If the provider's `on context changed` function terminates normally, associated `PROVIDER_CONTEXT_CHANGED` handlers MUST run.", (done) => {
+          const provider = new MockProvider({ initialStatus: ProviderStatus.READY });
+  
+          OpenFeature.setProvider(clientId, provider);
+          OpenFeature.setContext({});
+  
+          OpenFeature.addHandler(ProviderEvents.ContextChanged, () => {
```

Comment:
Expand the test to make sure the provider is included in the callback.

Reply:
done